Kubernetes default ingress

pasqualet opened the pull request:
**What type of PR is this?**

/kind bug
/kind feature

**What this PR does / why we need it**

Add an option in Kubernetes Cluster to create default IngressNginx instance.

**Which issue(s) this PR fixes**

Fixes #81 

**Special notes for your reviewer**

**Does this PR introduce a user-facing change?**
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below.
-->
```release-note
Add default IngressController in Kubernetes Cluster
Fixed calico race condition
Always redirect http to https in Kubernetes Cluster IngressController
```

bernardomp reviewed (changes requested):
La clases de Ingress funciona ok 
![image](https://user-images.githubusercontent.com/32247315/199832793-0d5ccaf9-e8cc-4a74-8d7a-9e9a7d1b4eaf.png)

Por otro lado,  parece que el parámetro public no funciona correctamente. Al menos, cuando se cambia a true o false,  no desencana ningun cambio a nivel de pulumi 

const cluster1 = new ct.Cluster("my-eks-ber",{networking: {

    admin: {
        domain: "apps.astrokube.es",
        public: false,
    },
    default: {
        domain: "default.astrokube.es",
        public: false,
    }

}});

pasqualet commented:
> La clases de Ingress funciona ok ![image](https://user-images.githubusercontent.com/32247315/199832793-0d5ccaf9-e8cc-4a74-8d7a-9e9a7d1b4eaf.png)
> 
> Por otro lado, parece que el parámetro public no funciona correctamente. Al menos, cuando se cambia a true o false, no desencadena ningún cambio a nivel de pulumi
> 
> ```
> const cluster1 = new ct.Cluster("my-eks-ber",{networking: {
>     admin: {
>         domain: "apps.astrokube.es",
>         public: false,
>     },
>     default: {
>         domain: "default.astrokube.es",
>         public: false,
>     }
> }});
> ```

You are right. I've fixed the issue with another commit. Anyway, even if we change the pulumi configuration, the LoadBalancer in AWS can't be changed. So it's a value only valid when the cluster is created. In the future 

> La clases de Ingress funciona ok ![image](https://user-images.githubusercontent.com/32247315/199832793-0d5ccaf9-e8cc-4a74-8d7a-9e9a7d1b4eaf.png)
> 
> Por otro lado, parece que el parámetro public no funciona correctamente. Al menos, cuando se cambia a true o false, no desencadena ningún cambio a nivel de pulumi
> 
> ```
> const cluster1 = new ct.Cluster("my-eks-ber",{networking: {
>     admin: {
>         domain: "apps.astrokube.es",
>         public: false,
>     },
>     default: {
>         domain: "default.astrokube.es",
>         public: false,
>     }
> }});
> ```

You are right. The public property wasn't propagated from the Cluster to the ClusterAddons component. Fixed in the new commit.

Anyway, even if Pulumi apply the changes, the LoadBalancer will not change to private/public because of AWS. So this can be configured only when the cluster is created, at least for now. In the future we have to recreate the IngressController component when this field changes.

pasqualet commented on provider/cmd/pulumi-resource-cloud-toolkit-aws/kubernetes/clusterAddons.ts:
```diff
@@ -71,6 +77,11 @@ export class ClusterAddons extends pulumi.ComponentResource {
    */
   public readonly awsLoadBalancerController: AwsLoadBalancerController;
 
+  public adminZoneArn?: pulumi.Input<string>;
```
Fixed!

bernardomp reviewed (commented):
OK, now pulumi detects changes and ingress class works.

![image](https://user-images.githubusercontent.com/32247315/200013668-9a273429-f459-472d-80be-fe489bcd8d6e.png)
